packages/components/src/components/text-input/_text-input.scss

```diff
@@ -156,7 +156,8 @@
     color: $disabled-02;
     background-color: $disabled-01;
     border-bottom: 1px solid transparent;
-
+    // Needed to fix disabled text in Safari #6673
+    -webkit-text-fill-color: currentColor;
```

Comment:
This the initial value, it's a noop: https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-text-fill-color.